src/transformStateWithClones.js

```diff
@@ -11,24 +11,30 @@ function transformStateWithClones(state, actions) {
   const history = [];
   const stateClone = Object.assign({}, state);
 
-  function iterate(arr) {
+  function deleteKeysFromObj(arr) {
     for (const key of arr) {
       delete stateClone[key];
     }
   }
 
   for (const action of actions) {
-    if (action.type === 'addProperties') {
-      Object.assign(stateClone, action.extraData);
-      history.push({ ...stateClone });
-    } else if (action.type === 'removeProperties') {
-      iterate(action.keysToRemove);
-      history.push({ ...stateClone });
-    } else if (action.type === 'clear') {
-      const keys = Object.keys(stateClone);
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateClone, action.extraData);
+        history.push({ ...stateClone });
+        break;
+      case 'removeProperties':
+        deleteKeysFromObj(action.keysToRemove);
+        history.push({ ...stateClone });
+        break;
+      case 'clear':
+        const keys = Object.keys(stateClone);
 
-      iterate(keys);
-      history.push({ ...stateClone });
+        deleteKeysFromObj(keys);
+        history.push({ ...stateClone });
+        break;
+      default:
+        throw Error('Error');
```

Comment:
For better user experience (very important in real apps) you should always specify the error, in this case you could say something like "invalid action type"